src/bench/bench.cpp

```diff
@@ -41,18 +42,47 @@ void GenerateTemplateResults(const std::vector<ankerl::nanobench::Result>& bench
 
 } // namespace
 
-benchmark::BenchRunner::BenchmarkMap& benchmark::BenchRunner::benchmarks()
+namespace benchmark {
+
+std::map<std::string, uint8_t> map_priority_level = {
+    std::make_pair("high", PriorityLevel::HIGH),
+    std::make_pair("medium", PriorityLevel::MEDIUM),
+    std::make_pair("low", PriorityLevel::LOW),
+    std::make_pair("all", 0xff)
+};
+
+struct CompPriorities {
+    template <typename T>
+    bool operator()(const T& l, const T& r) const {
+        return l.second < r.second;
+    }
+};
+
+std::string ListPriorities()
 {
-    static std::map<std::string, BenchFunction> benchmarks_map;
+    std::set<std::pair<std::string, uint8_t>, CompPriorities> vec(map_priority_level.begin(), map_priority_level.end());
```

Comment:
Since we're only using it in this function, I think I'd prefer a local lambda over a struct. Easier to read imo:

<details>
<summary>git diff</summary>

```diff
diff --git a/src/bench/bench.cpp b/src/bench/bench.cpp
index 13d3c203d..b9a8e85c4 100644
--- a/src/bench/bench.cpp
+++ b/src/bench/bench.cpp
@@ -51,17 +51,12 @@ std::map<std::string, uint8_t> map_priority_level = {
     std::make_pair("all", 0xff)
 };
 
-struct CompPriorities {
-    template <typename T>
-    bool operator()(const T& l, const T& r) const {
-        return l.second < r.second;
-    }
-};
-
 std::string ListPriorities()
 {
-    std::set<std::pair<std::string, uint8_t>, CompPriorities> vec(map_priority_level.begin(), map_priority_level.end());
-    return Join(vec, ',', [](const auto& entry){ return entry.first; });
+    using item_t = std::pair<std::string, uint8_t>;
+    auto sort_by_priority = [](item_t a, item_t b){ return a.second < b.second; };
+    std::set<item_t, decltype(sort_by_priority)> sorted_priorities(map_priority_level.begin(), map_priority_level.end(), sort_by_priority);
+    return Join(sorted_priorities, ',', [](const auto& entry){ return entry.first; });
 }
 
 uint8_t StringToPriority(const std::string& str)
```
</details>

If not, I think naming could be improved, e.g. `sorted_priorities` instead of `vec` (it's not a vector) and `SortByPriority` instead of `CompPriorities`?